IdentityCore/src/cache/accessor/MSIDAccountCredentialCache.m

```diff
@@ -315,8 +315,11 @@ - (BOOL)removeCredential:(nonnull MSIDCredentialCacheItem *)credential
     key.realm = credential.realm;
     key.target = credential.target;
     key.enrollmentId = credential.enrollmentId;
+    
+    // Append app key to service part if any in case this is an item in broker
+    MSIDCacheKey *overridenKey = [self appendAppKey:credential.appKey toKey:key];
```

Comment:
Just to discuss: should this be part of MSIDCredentialCacheKey instead?

Reply:
agree, addressed.